### DISCUSSION TITLE
Dynamic Suffix for the Route

### DISCUSSION THREAD

**Discussion by jvenki** [remix-run/remix#9574] [Q&A]
I understand that Remix allows one to define dynamic segment to match a particular route as given in https://remix.run/docs/en/main/file-conventions/routes#dynamic-segments. 

However, I want to match <<domain>>/gold-rate-*.html to be matching a particular route file. The * could point to any country or city and thus there are so many combinations. Unfortunately, I am not able to change the URL  for legacy reasons because there is already a google SEO score built for these URLs. How can I migrate to Remix and still ensure that all the urls like

<<domain>>/gold-rate-us.html
<<domain>>/gold-rate-jp.html
<<domain>>/gold-rate-in.html

all map to the same route file gold-rate-$place.tsx. 

Again please note that the wildcard is not a complete segment but part of a URL segment. Depending on the placeId, the route file will query a Database and then render the results

**Comment by sergiodxa** [comment]
You would need to make `gold-rate-us.html` the whole param, and validate in your loader that it has the correct shape, e.g. `routes/$slug.tsx` then in the loader use a RegEx like `/gold-rate-(\w+)\.html/` to grab the country code from the `params.slug`.

**Comment by jvenki** [comment]
Thanks for the prompt reply. 

The above was just a contrived example. In a legacy product which I am trying to migrate, there are lot of URLs of the form

``` 
/prefix-1-xx1.html
/prefix-1-xx2.html
....
/prefix-1-xxn.html
/prefix-2-xx1.html
/prefix-2-xx2.html
....
/prefix-2-xxn.html
```

Now prefix-1 has different features. prefix-2 has different features. I have already implemented what you suggested... but that is making all the browser bundles to be packaged into a single bundle which makes the bundle loftly. I understand why this has happened as it is a single route and  remix/vite has ended up doing it. 

Thats why I want to split into multiple routes for each prefix so that I am able to achieve code splitting properly.

I am just curious on why we couldnt support dynamic/regex based things on a URL segment when Remix React Router has solved all other things.

**Reply by sergiodxa** [reply]
> I am just curious on why we couldnt support dynamic/regex based things on a URL segment when Remix React Router has solved all other things

Because RR doesn't support matching more than one route to the same path segment, so you can only have one route and then branch inside your loaders and component based on the path segment value.
